src/processor/ValueSetProcessor.ts

```diff
@@ -14,14 +15,64 @@ export class ValueSetProcessor {
     }
   }
 
+  static extractRules(input: ProcessableValueSet, target: ExportableValueSet): void {
+    if (input.compose) {
+      input.compose.include?.forEach((vsComponent: any) =>
+        ValueSetProcessor.extractValueSetComponentRules(vsComponent, target, true)
+      );
+      input.compose.exclude?.forEach((vsComponent: any) =>
+        ValueSetProcessor.extractValueSetComponentRules(vsComponent, target, false)
+      );
+    }
+  }
+
+  static extractValueSetComponentRules(
+    vsComponent: any,
+    target: ExportableValueSet,
+    include: boolean
+  ) {
+    if (ValueSetProcessor.isValueSetComponent(vsComponent)) {
+      target.rules.push(ValueSetConceptComponentRuleExtractor.process(vsComponent, include));
+    }
+    target.rules = compact(target.rules);
+  }
+
   static process(input: any): ExportableValueSet {
     // We need something to call the ValueSet, so it must have a name or id
-    if (input.name != null || input.id != null) {
+    if (ValueSetProcessor.isProcessableValueSet(input)) {
       // Prefer name (which is optional), otherwise create a reasonable name from the id with only allowable characters
       const name = input.name ?? input.id.split(/[-.]+/).map(capitalize).join('');
       const valueSet = new ExportableValueSet(name);
       ValueSetProcessor.extractKeywords(input, valueSet);
       return valueSet;
     }
   }
+
+  // by FHIR spec, if the include list exists, it must contain at least one element
+  // but we can still do some processing without that.
+  // see http://hl7.org/fhir/r4/valueset-definitions.html#ValueSet.compose.include
+  static isProcessableValueSet(input: any): input is ProcessableValueSet {
+    return input.name != null || input.id != null;
+  }
+
+  static isValueSetComponent(input: any): input is ProcessableValueSetComponent {
+    return typeof input.system === 'string' || Array.isArray(input.valueSet);
+  }
```

Comment:
As noted above, I'm not sure it's such a bad thing if we process it even if there is no `input.system` or `input.valueSet`.  That should be very rare (since at least one is required by FHIR) and in the case that it does pop up, I think it's more appropriate to generate FSH that reflects the issue rather than pretend it didn't exist.  If skipping it really is the best option (because it really mucks with things in other code) then we should log an error.  But I think it's easier just to let the null value pass through as long as it doesn't wreak havoc elsewhere in GoFSH.

Reply:
Based on your 👍 , I take it that you _intended_ to address this comment?